cmd/clusterctl/client/init.go

```diff
@@ -209,6 +211,15 @@ func (c *clusterctlClient) setupInstaller(cluster cluster.Client, options InitOp
 		providerList:        providerList,
 	}
 
+	if !options.allowMissingProviderCRD {
+		providerList, err := cluster.ProviderInventory().List()
+		if err != nil {
+			return nil, err
+		}
+
+		addOptions.providerList = providerList
+	}
+
```

Comment:
@ykakarap I think this peace of code will only execute in the case of `Init` method. So, we don't want to be any missing CRD and it should return every possible error including missing CRD as well right? That's why skipped the `errors.Is` and returned the error directly.

```go
if !options.allowMissingProviderCRD {
    var noKindMatchErr *meta.NoKindMatchError
    providerList, err := cluster.ProviderInventory().List()
    if !errors.Is(err, noKindMatchErr) {
        return nil, err
    }

    addOptions.providerList = providerList
}
```
If I do like this it will assign a nil value of `providerList` in `addOptions.providerList` in case of missing CRD in `Init` operation and that will be a nil pointer issue.
Is there still any scope for improvement?

Reply:
This looks good. Good job! 
I tested it locally and it is working as expected.